test/ibmq/test_ibmq_qasm_simulator.py

```diff
@@ -124,14 +124,29 @@ def _new_submit(qobj, *args, **kwargs):
                              f"qobj header={qobj.header}")
             return mock.MagicMock()
 
-        backend = copy.deepcopy(self.sim_backend)
+        backend = self.sim_backend
+
+        sim_method = backend._configuration._data.get('simulation_method', None)
+        submit_fn = backend._submit_job
+
         backend._configuration._data['simulation_method'] = 'extended_stabilizer'
         backend._submit_job = _new_submit
 
-        circ = transpile(ReferenceCircuits.bell(), backend=backend)
-        backend.run(circ, header={'test': 'circuits'})
-        qobj = assemble(circ, backend=backend, header={'test': 'qobj'})
-        backend.run(qobj)
+        error = None
+
+        try:
+            circ = transpile(ReferenceCircuits.bell(), backend=backend)
+            backend.run(circ, header={'test': 'circuits'})
+            qobj = assemble(circ, backend=backend, header={'test': 'qobj'})
+            backend.run(qobj)
+        except QiskitError as err:
+            error = err
```

Comment:
You don't have to catch/re-raise the error. Doing
```
try:
    ...
finally:
    ....
```
will execute `finally` before raising any exceptions encountered in the try block.